### ISSUE TITLE
source_suffix containing dot leads to wrong source link

### ISSUE THREAD

**Issue by mgeier** [sphinx-doc/sphinx#7325]
**To Reproduce**

In `conf.py`, add:

```
source_suffix = {
    '.rst': 'restructuredtext',
    '.with.a.dot': 'restructuredtext',
}
```

Create a source file, e.g. `sourcefile.with.a.dot`, add it to `toctree`.

Run Sphinx.

The resulting page (`sourcefile.html`) will have a "Page source" link pointing to `_sources/sourcefile.dot.txt`. Note that part of the source suffix is missing.

**Expected behavior**

I would expect a source link to `_sources/sourcefile.with.a.dot.txt`.

**Your project**

A less contrived (but more complicated) example can be seen in https://github.com/spatialaudio/nbsphinx/pull/408.

**Environment info**
- Sphinx version: `master`

**Comment by tk0miya** [comment]
Fixed by #7328 
Thank you for reporting!

**Comment by mgeier** [comment]
Thanks for the quick fix, works like a charm!
